### PR TITLE
golangci lint

### PR DESCRIPTION
**Pull request by faddat**
- update make format and run make format, but don't fix lints
- first pass of new linter settings
- fix or mark lints as won't fix

the test for this code, is to see if it is possible to use this directly-- eg: take a testnet validator node, and run this branch's code.  It should work. 

If it doesn't the linting changed more than it ought to have.

### CODE REVIEW

**Comment by vishnukumavat** [comment]
This PR looks like the successor of #451, I would recommend closing #451 since the formatting changes made in 451 are also included in this PR.

**Comment by vishnukumavat** [comment]
Please update the PR with the development since #451 has been merged, this will help in a better review and will also avoid further occurring merge conflicts.
@faddat

File: x/asset/keeper/pairs_vault.go [hunk 1]
```diff
@@ -301,15 +301,15 @@ func (k Keeper) VaultIterateRewards(ctx sdk.Context, collectorLsr sdk.Dec, colle
...
 			var err error
 			if vaultData.BlockHeight == 0 {
-				interest, err = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, collectorBt)
+				_, _ = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, collectorBt)
 				if err != nil {
 					return
 				}
 			} else {
-				interest, err = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, vaultData.BlockTime.Unix())
+				_, _ = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, vaultData.BlockTime.Unix()) // NOTE: we might want to have this return an error but that would require chaging other code.
```
Comment (dheerajkd30) [inline x/asset/keeper/pairs_vault.go:314]: if an error is returned , the other vaultIDs won't be executed for the rewards. Better would be to log an error for faulty vaults.

File: x/asset/keeper/pairs_vault.go [hunk 2]
```diff
@@ -302,15 +302,16 @@ func (k Keeper) VaultIterateRewards(ctx sdk.Context, collectorLsr sdk.Dec, colle
 			if !found {
 				continue
 			}
+
 			var interest sdk.Dec
 			var err error
 			if vaultData.BlockHeight == 0 {
-				interest, err = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, collectorBt)
+				_, _ = k.rewards.CalculationOfRewards(ctx, vaultData.AmountOut, collectorLsr, collectorBt)
```
Comment (dheerajkd30) [inline x/asset/keeper/pairs_vault.go:309]: this change can't be accepted. interest, err has to be there else it breaks the below code and line 324 interest will always be the default value.

File: cmd/comdex/genaccounts.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
 		RunE: func(cmd *cobra.Command, args []string) error {
 			clientCtx := client.GetClientContextFromCmd(cmd)
 			depCdc := clientCtx.Codec
-			cdc := depCdc.(codec.ProtoCodecMarshaler)
+			cdc := depCdc.(codec.ProtoCodecMarshaler) //nolint:forcetypeassert // TODO: compare to eg: gaia and see if we can drop the type assertion.
```
Comment (dheerajkd30) [inline cmd/comdex/genaccounts.go:43]: this change has been included in different pr.

File: x/rewards/keeper/rewards.go [hunk 4]
```diff
@@ -578,12 +578,12 @@ func (k Keeper) CalculateLockerRewards(ctx sdk.Context, appID, assetID, lockerID
 	if !found {
 		return collectortypes.ErrorAssetDoesNotExist
 	}
-	rewards := sdk.ZeroDec()
+	rewards := sdk.ZeroDec() //nolint:ineffassign,staticcheck // might want to fix this, because there are a number of places where we are making vague declarations of new variables, then not using them till much later.
```
Comment (dheerajkd30) [inline x/rewards/keeper/rewards.go:581]: sure, will review this part and will refactor it.

**Comment by dheerajkd30** [comment]
stale PR

**Pull request closed without merging**